packages/kit/src/runtime/client/client.js

```diff
@@ -1548,6 +1570,10 @@ export function create_client({ target, base }) {
 						type: 'popstate',
 						delta
 					});
+
+					snapshots[current_history_index].forEach((value, i) => {
+						components[i]?.snapshot?.restore(value);
+					});
```

Comment:
Don't think this can happen here, if the navigation is blocked in `beforeNavigate`, this shouldn't run.

Reply:
good catch, wrapped in an `if (!blocked)` guard